### DISCUSSION TITLE
VSCode is not working

### DISCUSSION THREAD

**Discussion by thezzisu** [kimmknight/remoteapptool#137] [Q&A]
![image](https://github.com/user-attachments/assets/d02bf435-f9f8-4a15-bc96-1ee685f0eeee)
I have set up visual studio code on windows11 host. After generating RDP file and open it on client machine, vscode window opens in the host machine, not in the client machine, while a mstsc process is running in background on the client machine.

Is there any way to troubleshoot it?

**Comment by MrBrianGale** [comment]
Visual studio code is not an app you would want running in remote apps... Remote apps can't use hardware acceleration. 
A lot of apps don't support remote apps. Video games for example will mostly not work with remote apps. Pretty much any app that can spawn a child app will misbehave to. 

Not much you can do to make it work though as far as I'm aware.

**Comment by kimmknight** [comment]
I have used VS Code as a RemoteApp but not for a while. Following your question, I tested it again today:

![vscode-remoteapp-running](https://github.com/user-attachments/assets/926a99a1-6409-4194-9ad8-0bf508eac3c0)

(This is a Win 10 host and Win 10 client).

I am unclear on what hardware acceleration is available to RemoteApps since RemoteFX was deprecated. But in my case, the performance of VS Code seems fine for my purposes.

TBH, these days I wouldn't bother - GitHub Codespaces (fully portable, browser-based VSCode) is a better solution in many ways, however, I understand that many users of RemoteApp Tool have unique (often obscure) needs and scenarios.

In terms of troubleshooting, does it launch if connected to the host via a full-desktop session? Can you find any clues in the Windows Event Log?

**Reply by thezzisu** [reply]
In a full-desktop session vscode can work fine. (Both host and client is windows 11)
Also the weird thing is that vscode window opens in remote machine.
I will check for windows event log later.
